From a3abc4109c78dfbad851243c4609dd324d19eed2 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Wed, 14 Apr 2021 15:23:19 +0800
Subject: [PATCH 05/13] drivers: pci: s32gen1: fix build warning of unused
 function

The function s32gen1_pcie_config_irq() is used when CONFIG_PCI_S32GEN1_EP_MSI
or CONFIG_PCI_DW_DMA enabled, so add check line to fix the build warning.

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>

---
 drivers/pci/controller/dwc/pci-s32gen1.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/pci/controller/dwc/pci-s32gen1.c b/drivers/pci/controller/dwc/pci-s32gen1.c
index 0d32cf40cbce..1396a13c2d82 100644
--- a/drivers/pci/controller/dwc/pci-s32gen1.c
+++ b/drivers/pci/controller/dwc/pci-s32gen1.c
@@ -742,6 +742,7 @@ static struct dw_pcie_host_ops s32gen1_pcie_host_ops2 = {
 	.msi_host_init = s32gen1_pcie_msi_host_init,
 };
 
+#if defined(CONFIG_PCI_S32GEN1_EP_MSI) || defined(CONFIG_PCI_DW_DMA)
 #define MAX_IRQ_NAME_SIZE 32
 static int s32gen1_pcie_config_irq(int *irq_id, char *irq_name,
 		struct platform_device *pdev,
@@ -769,6 +770,7 @@ static int s32gen1_pcie_config_irq(int *irq_id, char *irq_name,
 
 	return 0;
 }
+#endif /* CONFIG_PCI_S32GEN1_EP_MSI || CONFIG_PCI_DW_DMA */
 
 static int s32gen1_add_pcie_port(struct pcie_port *pp,
 			struct platform_device *pdev)
-- 
2.25.1

